src/Assembly/Helpers/Net/Sockets/ClientPokeSessionManager.cs

```
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Net;
+using System.Diagnostics;
+using Assembly.Metro.Dialogs;
+using System.ComponentModel;
+
+namespace Assembly.Helpers.Net.Sockets
+{
+    public class ClientPokeSessionManager : IPokeSessionManager
+    {
+        private NetworkPokeClient _client;
+        private IPokeCommandHandler _handler;
+        private bool _sessionActive;
+
+        public event EventHandler SessionActive;
+        public event EventHandler<RunWorkerCompletedEventArgs> SessionDead;
+
+        public ClientPokeSessionManager(IPokeCommandHandler handler)
+        {
+            _client = new NetworkPokeClient();
+            _handler = handler;
+        }
+
+        public bool StartClient(IPEndPoint endpoint)
+        {
+            try
+            {
+                _client.Connect(endpoint);
+            }
+            catch (Exception)
```

Not a fan of hiding the exception here.